formula-android/src/main/java/com/instacart/formula/integration/FragmentFlowRenderView.kt

```diff
@@ -198,9 +198,28 @@ internal class FragmentFlowRenderView(
     }
 
     private fun removeFragment(fragment: Fragment) {
-        onFragmentViewStateChanged(fragment.getFragmentKey(), false)
+        onFragmentViewStateChanged(fragment.getFormulaFragmentId(), false)
         val event = FragmentLifecycle.createRemovedEvent(fragment)
         onLifecycleEvent(event)
-        removedEarly.add(fragment.getFragmentKey())
+        removedEarly.add(fragment.getFormulaFragmentId())
+    }
+
+    /**
+     * Creates a unique identifier the first time fragment is attached that
+     * is persisted across configuration changes.
+     */
+    private fun initializeFragmentInstanceIdIfNeeded(f: Fragment) {
+        if (f is BaseFormulaFragment<*>) {
+            val arguments = f.arguments ?: run {
+                Bundle().apply {
+                    f.arguments = this
+                }
+            }
+            val id = arguments.getString(FormulaFragment.ARG_FORMULA_ID, "")
+            if (id.isNullOrBlank()) {
+                val initializedId = UUID.randomUUID().toString()
```

Comment:
Maybe lookup for existing fragments `ARG_FORMULA_ID`, to avoid collision?

Reply:
This is an interesting thought which I had to read up more on.

From [Wikipedia](https://en.wikipedia.org/wiki/Universally_unique_identifier#Random_UUID_probability_of_duplicates)
```
While the probability that a UUID will be duplicated is not zero, it is close enough to zero to be negligible
```

Also, from [Stackoverflow](https://stackoverflow.com/questions/24876188/how-big-is-the-chance-to-get-a-java-uuid-randomuuid-collision)

<img width="693" alt="Screen Shot 2021-04-26 at 11 29 03 AM" src="https://user-images.githubusercontent.com/2983787/116132673-a8082f00-a682-11eb-8949-2844d009957b.png">


Also, since we use this in-combination with `FragmentKey`, it's even less change for a duplicate.

Reply:
This 1 crash in 100 years from now is going to be on you 🤣